### PR TITLE
Form

### PR DESCRIPTION
**Pull request by denizulku**
Please have a look into Form, List & Habit as they all tie into the Form component
This PR refers to issue #6

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
 import styled from 'styled-components'
+import Form from './components/Form/Form'
 
 function App() {
   return (
     <div className="App">
-      <header>
-        <h1>Habit Tracker</h1>
-      </header>
+      <header></header>
```
Comment (karstenlenz) [inline src/App.js]: Do you really need the empty  `<header>` ?

File: src/components/Form/Form.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import React, { useState } from 'react'
+import styled from 'styled-components'
+import List from '../List/List'
+
+export default function Form() {
+  const [habits, setHabits] = useState([])
+  return (
+    <>
+      <Container onSubmit={handleSubmit}>
```
Comment (karstenlenz) [inline src/components/Form/Form.js]: I think the name "container" is a bit too generic. I wondered why a container has an "onSubmit" attribute. Maybe rename it to something like "formContainer" ?

File: src/components/Form/Form.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    form.reset()
+  }
+}
+
+const Container = styled.form`
+  margin-top: 20px;
+  display: flex;
+  justify-content: center;
+`
+const Input = styled.input`
```
Comment (karstenlenz) [inline src/components/Form/Form.js]: I think you shouldn't use the same name for HTML-Elements like "input". Maybe call it "InputStyled" if you can't find a name that makes more sense in this context.

File: src/components/Form/Form.test.js [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import React from 'react'
+import { render } from '@testing-library/react'
+import Form from './Form.js'
+
+test('the form data is correct', () => {
```
Comment (karstenlenz) [inline src/components/Form/Form.test.js]: You are only doing a snapshot test, but aren't really testing if the form data is correct. If would change the description to something like "renders correctly".

File: src/components/Habit/HabitItem.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import React from 'react'
+
+export default function Habit({ text, completed }) {
```
Comment (karstenlenz) [inline src/components/Habit/HabitItem.js:4]: You could add PropTypes if you like for your components to make sure all the props are used correctly.

File: src/components/List/List.js [hunk 6]
```diff
@@ -0,0 +1,32 @@
+import React from 'react'
+import styled from 'styled-components'
+import Habit from '../Habit/HabitItem'
+
+export default function List({ habits }) {
+  return (
+    <HabitList>
+      {habits.map((habit, completed) => (
```
Comment (karstenlenz) [inline src/components/List/List.js]: Does this map-function work correctly? From what I understand the second paramameter in map callback should be the index, not the completed property.
see https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Global_Objects/Array/map

**Review by karstenlenz — COMMENTED** [review]
Nice work! I added a few minor comments.

File: src/components/Form/Form.js [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+  }
+}
+
+const FormContainer = styled.form`
+  margin-top: 20px;
+  display: flex;
+  justify-content: center;
+`
+const InputStyled = styled.input`
+  width: 300px;
```
Comment (manuelpapa) [inline src/components/Form/Form.js:34]: Maybe think of relative values `rem` to provide a better experience on different screen sizes.

File: src/components/List/List.js [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+  margin-top: 20px;
+  position: relative;
+  text-align: center;
+`
+const HabitContainer = styled.div`
+  display: inline-block;
+  margin-top: 10px;
+  width: 260px;
+  height: 25px;
+  background-color: #ffc179;
```
Comment (manuelpapa) [inline src/components/List/List.js:30]: In a later step you can think of creating `$variables` for the colors in your `GlobalStyles.js` and then just import them as e.g. `color: var(--main-color);`.

**Review by manuelpapa — COMMENTED** [review]
GJ

**Pull request merged**